cloud/k8s/deploy/kontain-deploy/overlays/kkm/cm-install-lib-kkm.yaml

```diff
@@ -22,25 +22,33 @@ metadata:
   labels:
     app: kontain-init
 data:
-  kontain_install_lib.sh: |-
+  kontain_install_lib.sh: >
     #!/usr/bin/env bash
 
-    # This will get overridden be ENV definition in kontain-deploy.yaml
-    KONTAIN_RELEASE_URL="${KONTAIN_RELEASE_URL:-https://github.com/kontainapp/km/releases/download/current/kontain_bin.tar.gz}"
+    TAG=${1:-$(curl -L -s https://raw.githubusercontent.com/kontainapp/km/current/km-releases/current_release.txt)}
+    KONTAIN_RELEASE_URL="https://github.com/kontainapp/km/releases/download/${TAG}/kontain_bin.tar.gz"
+    echo KONTAIN_RELEASE_URL is: ${KONTAIN_RELEASE_URL}
 
-    # we use chroot here because the host is rooted at /root
-    # and kkm install looks for sudo and dkms in locations on the host
     function install_kontain_artifacts() {
-
       ROOT_MOUNT_DIR="${ROOT_MOUNT_DIR:-/root}"
 
-      # check to see if deltarpm is installed and if not install it (for Amzn Linux 2)
-      yum -q list installed deltarpm &>/dev/null && echo "deltarpm already installed" || (echo "Not installed"; yum install -y deltarpm)
+      source /$ROOT_MOUNT_DIR/etc/os-release
+      if [[ "$ID" == "amzn" ]]; then 
+        echo we are running this daemonset on ${ID}
+        chroot ${ROOT_MOUNT_DIR} rm /var/lib/rpm/__db*
+        chroot ${ROOT_MOUNT_DIR} yum update -y
+        sleep 2
+        chroot ${ROOT_MOUNT_DIR} yum install  -y kernel-devel-$(uname -r)
+        sleep 2
+        chroot ${ROOT_MOUNT_DIR} ls /lib/modules
```

Comment:
This can probably be removed eventually but I use it for debugging when things fail.  Because several times, I have found that the script can fail when the headers (especfially on AWS amazon linux 2) can either be not present or be of the wrong version, leading to failure in running kkm.run.  This allows me to see the output when using: 

$ kubectl logs -f <daemonset-pod>

Reply:
Got your point on chroot vs direct consistency.  Will revise that uniformity, when I revisit this next.